src/math/bandmat.rs

```diff
@@ -0,0 +1,170 @@
+// Copyright 2015 Michael Yang. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+#![macro_use]
+
+use libc::c_int as c_int;
+use num::traits::NumCast;
+use std::fmt;
+use std::iter::repeat;
+use std::ops::Index;
+use std::slice;
+use vector::ops::Copy;
+use matrix::BandMatrix;
+use Matrix;
+
+#[derive(Debug, PartialEq)]
+pub struct BandMat<T> {
+    rows: usize,
+    cols: usize,
+    sub_diagonals: c_int,
+    sup_diagonals: c_int,
+    data: Vec<T>,
+}
+
+impl<T> BandMat<T> {
+    pub fn new(n: usize, m: usize, sub: c_int, sup: c_int) -> BandMat<T> {
+        let len = n * m;
+        let mut data = Vec::with_capacity(len);
+        unsafe {
+            data.set_len(len);
+        }
+
+        BandMat {
+            rows: n,
+            cols: m,
+            data: data,
+            sub_diagonals: sub,
+            sup_diagonals: sup,
+        }
+    }
+
+    pub fn rows(&self) -> usize {
+        self.rows
+    }
+    pub fn cols(&self) -> usize {
+        self.cols
+    }
+    pub unsafe fn set_rows(&mut self, n: usize) {
+        self.rows = n;
+    }
+    pub unsafe fn set_cols(&mut self, n: usize) {
+        self.cols = n;
+    }
+
+    pub unsafe fn set_sub_diagonals(&mut self, n: c_int) {
```

Comment:
do we want to expose `c_int` in public API? How is it done in the rest of the API?

Reply:
`c_int` is just a wrapper for `i32`, but perhaps we should standardize to `usize` or `u32` since having negative sub and super diagonals doesn't make mathematical sense?

Reply:
I fully agree